clustermesh-apiserver/syncstate/syncstate_test.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package syncstate
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Configure a generous timeout to prevent flakes when running in a noisy CI environment.
+var (
+	tick    = 10 * time.Millisecond
+	timeout = 5 * time.Second
+)
+
+func TestSyncState(t *testing.T) {
+	var doneFuncs []func(context.Context)
+	ss := new(MetricsProvider()())
+
+	// add several resource to the SyncState
+	for i := 0; i < 3; i++ {
+		doneFuncs = append(doneFuncs, ss.WaitForResource())
+	}
+	ss.Stop()
+
+	// call all doneFuncs to simulate the resources being synchronized
+	for _, f := range doneFuncs {
+		f(context.Background())
+	}
+
+	require.EventuallyWithTf(t, func(c *assert.CollectT) {
+		assert.True(c, ss.Complete())
+		_, ok := <-ss.WaitChannel()
+		assert.False(c, ok)
+	}, timeout, tick, "Expected SyncState to be completed")
```

Comment:
Could be tested synchronously (without `EnvetuallyWithTf`), as it doesn't depend on any background work. The two assertions look also redundant to me, as testing the same aspect.